pallets/asset-handler/src/tests.rs

```diff
@@ -589,46 +591,103 @@ pub fn test_convert_amount_for_foreign_chain() {
-			Origin::signed(1),
-			0,
-			5,
-			BoundedVec::try_from(asset_address.to_fixed_bytes().to_vec()).unwrap()
-		));
-
-		let mut derived_asset_id = vec![];
-		derived_asset_id.push(0);
-		derived_asset_id.push(5);
-		derived_asset_id.extend(&asset_address.to_fixed_bytes()[0..5]);
-
-		// Hash the resulting vector with Keccak256 Hashing Algorithm and retrieve first 16 bytes
-		let derived_asset_id_hash = &sp_io::hashing::keccak_256(derived_asset_id.as_ref())[0..16];
+	let asset_address: H160 = ASSET_ADDRESS.parse().unwrap();
+	let asset_id = create_thea_asset_id(0, 5);
 
-		// Derive u128 from resulting bytes
-		let mut temp = [0u8; 16];
-		temp.copy_from_slice(derived_asset_id_hash);
-		let asset_id = u128::from_le_bytes(temp);
+	new_test_ext().execute_with(|| {
+		assert_ok!(create_thea_asset(asset_address, 0, 5));
 
 		let (network_id, identifier_length, identifier) = <TheaAssets<Test>>::get(asset_id);
 
 		// Assert Storage
 		assert_eq!(network_id, 0);
 		assert_eq!(identifier_length, 5);
 		assert_eq!(identifier.to_vec(), asset_address.to_fixed_bytes().to_vec());
+	})
+}
+
+#[test]
+pub fn test_create_thea_asset_with_mismatching_identifier_will_return_identifier_length_mismatch_error(
+) {
+	// TODO: Implement
+}
+
+#[test]
+pub fn test_mint_thea_asset_with_unknown_recipient_will_return_cannot_create_error() {
+	let asset_address: H160 = ASSET_ADDRESS.parse().unwrap();
+	let asset_id = create_thea_asset_id(0, 5);
 
-		// mint into account
+	new_test_ext().execute_with(|| {
+		assert_ok!(create_thea_asset(asset_address, 0, 5));
 		assert_noop!(
 			AssetHandler::mint_thea_asset(asset_id, u64::MAX, 1_000_000_000_000_0_u128),
 			TokenError::CannotCreate
 		);
 	})
 }
 
+#[test]
+pub fn test_mint_thea_asset_with_not_registered_asset_will_return_asset_not_registered_error() {
+	let recipient = create_recipient_account();
+	let asset_id = create_thea_asset_id(0, 5);
+
+	new_test_ext().execute_with(|| {
+		assert_noop!(
+			AssetHandler::mint_thea_asset(asset_id, recipient, 1_000_000_000_000_0_u128),
+			Error::<Test>::AssetNotRegistered
+		);
+	})
+}
+
+#[test]
+pub fn test_mint_thea_asset_with_zero_amount_will_return_amount_cannot_be_zero_error() {
+	let asset_address: H160 = ASSET_ADDRESS.parse().unwrap();
+	let recipient = create_recipient_account();
+	let asset_id = create_thea_asset_id(0, 5);
+
+	new_test_ext().execute_with(|| {
+		assert_ok!(create_thea_asset(asset_address, 0, 5));
+		assert_noop!(
+			AssetHandler::mint_thea_asset(asset_id, recipient, 0_u128),
+			Error::<Test>::AmountCannotBeZero
+		);
+	})
+}
+
+#[test]
+pub fn test_mint_thea_asset_will_increase_asset_balance() {
+	let asset_address: H160 = ASSET_ADDRESS.parse().unwrap();
+	let recipient = create_recipient_account();
+	let asset_id = create_thea_asset_id(0, 5);
+
+	new_test_ext().execute_with(|| {
+		assert_ok!(create_thea_asset(asset_address, 0, 5));
+		// FIXME: Investigate why `mint_thea_asset` from the next line is returns
```

Comment:
This is expected behavior, Since the account `10` does not have any existential deposit. Mint function will return `TokeError::CannotCreate` meaning account cannot be created. So we will have to `set balance` first using `Balances` pallet to the account `10` and then call `mint_thea_asset`.